requests/utils.py

```
@@ -216,6 +218,38 @@ def guess_filename(obj):
+
+    # find the first valid part of the provided path and treat that as a zip archive
+    # assume the rest of the path is the name of a member in the archive
+    archive, member = os.path.split(path)
+    while archive and not os.path.exists(archive):
+        archive, prefix = os.path.split(archive)
+        member = '/'.join([prefix, member])
+
+    if not zipfile.is_zipfile(archive):
+        return path
+
+    zip_file = zipfile.ZipFile(archive)
```

Pylint (3.3.7)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)